bib/cmd/bootc-image-builder/image.go

```diff
@@ -251,6 +252,29 @@ func genPartitionTable(c *ManifestConfig, customizations *blueprint.Customizatio
 		}
 	}
 
+	if c.SourceInfo.KernelInfo.HasAbootImg {
+		var abootPartition *disk.Partition
+
+		sourcePipeline := "build"
+		if c.BuildSourceInfo != nil {
+			sourcePipeline = "target"
+		}
+
+		for i := range partitionTable.Partitions {
```

Comment:
(nitpick) this could be written slightly more compact using the slices package as:
```go
	if c.SourceInfo.KernelInfo.HasAbootImg {
		idx := slices.IndexFunc(partitionTable.Partitions, func(part disk.Partition) bool {
			return part.Label == "ukiboot_a" || part.Label == "aboot_a"
		})
		if idx >= 0 {
			sourcePipeline := "build"
			if c.BuildSourceInfo != nil {
				sourcePipeline = "target"
			}

			partitionTable.Partitions[idx].Payload = &disk.Raw{
				SourcePipeline: sourcePipeline,
				SourcePath:     filepath.Join("/usr/lib/modules/", c.SourceInfo.KernelInfo.Version, "aboot.img"),
			}
		}
``` 
but of course a bit of a matter of preference, feel free to keep as is if you prefer it that way.